influxdb/1.2/alpine/entrypoint.sh

```diff
@@ -1,8 +1,128 @@
-#!/bin/sh
+#!/bin/bash
 set -e
 
 if [ "${1:0:1}" = '-' ]; then
     set -- influxd "$@"
 fi
 
-exec "$@"
+AUTH_ENABLED="$INFLUXDB_HTTP_AUTH_ENABLED"
+
+if [ -z "$AUTH_ENABLED" ]; then
+	AUTH_ENABLED="$(grep -iE '^\s*auth-enabled\s*=\s*true' /etc/influxdb/influxdb.conf | grep -io 'true' | cat)"
```

Comment:
I would prefer if we didn't have an alternative environment variable for this. If we want to change the environment variable, then setting `AUTH_ENABLED` should just ensure that `INFLUXDB_HTTP_AUTH_ENABLED` is set to true rather than rewriting the configuration file. Especially since this configuration file could be provided by the user and I don't want to rewrite someone's configuration file.